frontend/hub/access/roles/RolePage/HubRoleForm.tsx

```diff
@@ -0,0 +1,177 @@
+import { useNavigate, useParams } from 'react-router-dom';
+import { useWatch } from 'react-hook-form';
+import {
+  PageFormSelect,
+  PageFormSubmitHandler,
+  PageFormTextInput,
+  PageHeader,
+  PageLayout,
+  useGetPageUrl,
+  usePageNavigate,
+} from '../../../../../framework';
+import { useInvalidateCacheOnUnmount } from '../../../../common/useInvalidateCache';
+import { usePostRequest } from '../../../../common/crud/usePostRequest';
+import { HubRbacRole } from '../../../interfaces/expanded/HubRbacRole';
+import { hubAPI } from '../../../common/api/formatPath';
+import { HubRoute } from '../../../main/HubRoutes';
+import { HubPageForm } from '../../../common/HubPageForm';
+import { useHubRoleMetadata } from '../hooks/useHubRoleMetadata';
+import { PageFormHidden } from '../../../../../framework/PageForm/Utils/PageFormHidden';
+import { PageFormMultiSelect } from '../../../../../framework/PageForm/Inputs/PageFormMultiSelect';
+import { ContentTypeEnum } from '../../../interfaces/expanded/ContentType';
+import { useGet } from '../../../../common/crud/useGet';
+import { usePatchRequest } from '../../../../common/crud/usePatchRequest';
+import { useIsValidRoleName } from '../hooks/useIsValidRoleName';
+
+export function CreateRole(props: { breadcrumbLabelForPreviousPage?: string }) {
+  const { t } = useTranslation();
+  const navigate = useNavigate();
+  const pageNavigate = usePageNavigate();
+
+  useInvalidateCacheOnUnmount();
+
+  const postRequest = usePostRequest<Partial<HubRbacRole>, HubRbacRole>();
+
+  const onSubmit: PageFormSubmitHandler<HubRbacRole> = async (Role) => {
+    const newRole = await postRequest(hubAPI`/_ui/v2/role_definitions/`, Role);
+    pageNavigate(HubRoute.RolePage, { params: { id: newRole.id } });
+  };
+  const onCancel = () => navigate(-1);
+  const getPageUrl = useGetPageUrl();
+
+  return (
+    <PageLayout>
+      <PageHeader
+        title={t('Create role')}
+        breadcrumbs={[
+          {
+            label: props.breadcrumbLabelForPreviousPage || t('Roles'),
+            to: getPageUrl(HubRoute.Roles),
+          },
+          { label: t('Create role') },
+        ]}
+      />
+      <HubPageForm<HubRbacRole>
+        submitText={t('Create role')}
+        onSubmit={onSubmit}
+        cancelText={t('Cancel')}
+        onCancel={onCancel}
+      >
+        <HubRoleInputs />
+      </HubPageForm>
+    </PageLayout>
+  );
+}
+
+export function EditRole(props: { breadcrumbLabelForPreviousPage?: string }) {
+  const { t } = useTranslation();
+  const navigate = useNavigate();
+  const pageNavigate = usePageNavigate();
+  const params = useParams<{ id?: string }>();
+  const id = Number(params.id);
+  const { data: Role } = useGet<HubRbacRole>(hubAPI`/_ui/v2/role_definitions/${id.toString()}/`);
+
+  useInvalidateCacheOnUnmount();
+
+  const postRequest = usePostRequest<Partial<HubRbacRole>, HubRbacRole>();
+  const patchRequest = usePatchRequest<Partial<HubRbacRole>, HubRbacRole>();
+
```

Comment:
When attempting to edit a role already existent, but editable.  Just a few field should be editable, basically name and description. But generates an error.


<img width="1727" alt="image" src="https://github.com/user-attachments/assets/4dc5f167-393f-4291-8729-fa4dd6da3f48">

Reply:
There is an API bug [AAH-3348](https://issues.redhat.com/browse/AAH-3348). If this role is built-in it should not be editable. However the API is returning the wrong value of the `managed` flag so the UI thinks it is editable.